shoop/themes/classic_gray/templates/classic_gray/shoop/front/basket/default_basket.jinja

```diff
@@ -41,6 +41,39 @@
         $("#update_basket_form .quantity :input").change(function() {
             $("#update_basket_form").submit();
         });
+
+        $("#submit-code").click(function(e){
+            e.preventDefault();
+            const coupon = $("#discount-code").val();
+
+            if (coupon.length) {
+                // ajax request to command
+                const data = {
+                    "command": "add_campaign_code",
+                    "code": coupon
+                };
+
+                $.ajax({
+                    url: "",
+                    method: "POST",
+                    data: data,
+                    success: function(response) {
+                        if (response.ok) {
+                            location.reload();
+                        }
+                        else {
+                            alert(gettext("Coupon is not available."))
```

Comment:
Hmm you shouldn't probably use gettext inside jinja file. Why not just use default trans-tags?